raft/src/test/java/org/apache/kafka/raft/internals/RecordsIteratorTest.java

```diff
@@ -98,8 +98,12 @@ public void testCrcValidation(
     ) throws IOException {
         List<TestBatch<String>> batches = createBatches(seed);
         MemoryRecords memRecords = buildRecords(compressionType, batches);
-        // Corrupt the record buffer
-        memRecords.buffer().putInt(DefaultRecordBatch.LAST_OFFSET_DELTA_OFFSET, new Random(seed).nextInt());
+        // Read the Batch CRC for the first batch from the buffer
+        ByteBuffer readBuf = memRecords.buffer().asReadOnlyBuffer();
```

Comment:
Btw, `adReadOnlyBuffer` is probably not needed since `MemoryBuffer#buffer` returns a duplicate view into the buffer.